Cue/app/tabs/library/play/PlayDeckSetupView.js

```diff
@@ -37,23 +39,88 @@ export default class PlayDeckSetupView extends React.Component {
 
   state: {
     playbackOption: PlaybackOption,
-    customStartIndex: number
-  }
-
-  _onPlaybackOptionSelected = (option: PlaybackOption) => {
-    this.setState({
-      ...this.state,
-      playbackOption: option
-    })
+    customStartIndex: number,
+    flaggedOnly: boolean,
+    answerFirst: boolean,
+    filteredCards: Array<Card>,
   }
 
   constructor(props: Props) {
     super(props)
 
-    this.state = {
+    let state = {
       playbackOption: 'sequential',
-      customStartIndex: 0
+      flaggedOnly: this.props.flagFilter,
+      answerFirst: false,
+      customStartIndex: 0,
+    }
+    state.filteredCards = this._generateDeckFilter(state)(this.props.deck.cards || [])
+    this.state = state
+  }
+
+  _flaggedOnly(array: Array<Card>): Array<Card> {
+    return array.filter(card => card.needs_review)
+  }
+
+  _answerFirst(array: Array<Card>): Array<Card> {
+    return array.map(card => {return {...card, front: card.back, back: card.front}})
+  }
+
+  _shuffled(array: Array<Card>): Array<Card> {
+    let ret = array.slice(0)
+
+    for (let i = ret.length - 1; i > 0; i--) {
+      let j = Math.floor(Math.random() * (i + 1))
+      let temp = ret[i]
+      ret[i] = ret[j]
+      ret[j] = temp
+    }
+
+    return ret
+  }
+
+  _generateDeckFilter(state) {
+    return (array: Array<Card>): Array<Card> => {
+      let ret = array.slice()
+
+      if (state.flaggedOnly) {
+        ret = this._flaggedOnly(ret)
+      }
+      if (state.answerFirst) {
+        ret = this._answerFirst(ret)
+      }
+      if (state.playbackOption === 'shuffled') {
+        ret = this._shuffled(ret)
+      }
+
+      return ret
+    }
+  }
+
+  _setFilterOption(option) {
+    let newState = {
+      ...this.state,
+      ...option,
+      customStartIndex: 0,
     }
+    newState.filteredCards = this._generateDeckFilter(newState)(this.props.deck.cards || [])
+
+    this.setState(newState)
+  }
+
+  _onPressPlay = () => {
+    if (this.state.filteredCards.length === 0) {
+      Alert.alert(
+        Platform.OS === 'android' ? 'No cards to play' : 'No Cards To Play',
+        'Try turning off the "Flagged cards only" option'
+      )
+      return
+    }
+    this.props.navigator.replace({
+      playDeck: this.props.deck,
+      cardFilter: this._generateDeckFilter(this.state),
```

Comment:
This is kinda weird but I guess it makes sense since we need access to other properties from the deck in `PlayDeckView`.